bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -205,35 +205,62 @@ public BlackboardArtifact.Type getOrAddArtifactType(String typeName, String disp
 			return typeNameToArtifactTypeMap.get(typeName);
 		}
 
-		Statement s = null;
-		ResultSet rs = null;
 		CaseDbTransaction trans = null;
 		try {
 			trans = caseDb.beginTransaction();
 
 			CaseDbConnection connection = trans.getConnection();
-			s = connection.createStatement();
-			rs = connection.executeQuery(s, "SELECT artifact_type_id FROM blackboard_artifact_types WHERE type_name = '" + typeName + "'"); //NON-NLS
-			if (!rs.next()) {
-				rs.close();
-				rs = connection.executeQuery(s, "SELECT MAX(artifact_type_id) AS highest_id FROM blackboard_artifact_types");
+
```

Comment:
Done we need a readlock before this?

Reply:
SleuthkitCase.beginTransaction() gets a write lock itself